MarioDoubleEvolution/src/main/kotlin/cz/cuni/mff/aspect/evolution/Fitness.kt

```diff
@@ -7,18 +7,16 @@ import cz.cuni.mff.aspect.mario.controllers.MarioController
 import cz.cuni.mff.aspect.mario.level.MarioLevel
 
 
-typealias Fitness = (controller: MarioController, levels: Array<MarioLevel>) -> Float
+// TODO: rename me
+typealias Fitness<F> = (gameStatistics: Array<GameStatistics>) -> F
 
 
-fun fitnessOnlyDistance(controller: MarioController, levels: Array<MarioLevel>): Float {
-    val statistics = playLevels(controller, levels)
+fun fitnessOnlyDistance(statistics: Array<GameStatistics>): Float {
```

Comment:
wrap these into some object and also don't use fitness in their names because they can be used as objective function too